ceph/rados/perf_workflow.py

```diff
@@ -0,0 +1,160 @@
+"""
+Module to perform specific functionalities needed to run
+Performance Benchmarking with RADOS
+"""
+
+import time
+from math import floor
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from utility.log import Log
+from utility.utils import method_should_succeed
+
+log = Log(__name__)
+
+
+class PerfWorkflows:
+    """
+    Contains various functions to run performance benchmarking at rados layer
+    """
+
+    def __init__(self, node: CephAdmin):
+        """
+        initializes the env to run Perf evaluation for rados
+        Args:
+            node: CephAdmin object
+        """
+        self.rados_obj = RadosOrchestrator(node=node)
+        self.cluster = node.cluster
+        self.clients = node.cluster.get_nodes(role="client")
+
+    def create_pool_per_config(self, **pool_config):
+        """
+        Method to create required pools at the start of test
+        Args:
+            pool_config: config specific to the pool
+        Returns:
+            True -> pass, False -> fail
+        """
+        # Creating pools
+        log.info(
+            f"Creating {pool_config['pool_type']} pool on the cluster with name {pool_config['pool_name']}"
+        )
+
+        # setting pg_num_min to pg_num so that pools are created with required
+        # number of PGs from the start
+        pool_config["pg_num_min"] = pool_config["pg_num"]
+
+        if pool_config.get("pool_type", "replicated") == "erasure":
+            method_should_succeed(
+                self.rados_obj.create_erasure_pool,
+                pool_config["pool_name"],
+                **pool_config,
+            )
+        else:
+            method_should_succeed(self.rados_obj.create_pool, **pool_config)
+
+        log.info(f"Pool {pool_config['pool_name']} created successfully")
+        return True
+
+    def get_object_count(self, fill_percent, pool_name, obj_size, num_client) -> int:
+        """
+        Method to calculate how many objects should be written of each size
+        Args:
+            fill_percent(int): pool fill percentage
+            pool_name(str): name of the pool
+            obj_size(str): object size
+            num_client(int): number of clients
+        Returns:
+            obj count to be written by the client
+        """
+        # total available space for the pool
+        pool_stat = self.rados_obj.get_cephdf_stats(pool_name=pool_name)
+        max_avail_kb = pool_stat["stats"]["max_avail"]
+        # max_avail_mb = floor(max_avail_kb / 1 << 10)
+
+        # amount of data to be written to the pool
+        total_write_data = floor(max_avail_kb * fill_percent / 100)
```

Comment:
Are we taking into consideration the replica size? for calculating total amount of data to be written?

Reply:
If for now, we are only targeting RE pools, we can add that in a doc string. It's good enough to start from RE pools

Reply:
The pool multiplier or maybe I should rename it to pool divider has been written to take care of replica size and facilitate object count calculation accordingly